### PR TITLE
Add `useCameraRoll()` hook

### PR DESCRIPTION
**Pull request by retyui**
# Summary

See: https://github.com/react-native-community/hooks/issues/267#issuecomment-947424510

## Test Plan

```
yarn test
```

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added the documentation in `README.md`
- [x] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

**Comment by retyui** [comment]
Hi @bartolkaruza @idrissakhi 

could you please review my PR?

**Review by santhoshvai — APPROVED** [review]

**Comment by santhoshvai** [comment]
@Arjan-Zuidema can you please review this too

File: src/CameraRoll.ts [hunk 1]
```diff
@@ -238,3 +239,51 @@ export class CameraRoll {
...
+    } catch (error) {
+      if (__DEV__) console.log('[useCameraRoll] Error getting photos: ', error);
+    }
+  }
+
+  async function saveToCameraRoll(
+    ...args: Parameters<typeof CameraRoll.saveToCameraRoll>
+  ): Promise<void> {
+    try {
+      await CameraRoll.saveToCameraRoll(...args);
```
Comment (Arjan-Zuidema) [inline src/CameraRoll.ts:281]: Could you change this to use the `save` function and not the deprecated `saveToCameraRoll` instead?

**Review by Arjan-Zuidema — CHANGES_REQUESTED** [review]
If you could fix this one thing I commented it can be merged

File: src/CameraRoll.ts (hunk 1 above)
Comment (davydspotnana) [reply src/CameraRoll.ts]: done

Comment (retyui) [reply src/CameraRoll.ts]: done

File: src/CameraRoll.ts [hunk 2]
```diff
@@ -238,3 +240,51 @@ export class CameraRoll {
...
+  ): Promise<void> {
+    try {
+      const result = await CameraRoll.getPhotos(config);
+      setPhotos(result);
+    } catch (error) {
+      if (__DEV__) console.log('[useCameraRoll] Error getting photos: ', error);
+    }
+  }
+
+  async function saveToCameraRoll(
```
Comment (Arjan-Zuidema) [inline src/CameraRoll.ts:278]: Maybe also rename this to just `save`, sorry for being nitpicky. Otherwise it looks good to me

**Review by Arjan-Zuidema — CHANGES_REQUESTED** [review]

File: src/CameraRoll.ts (hunk 2 above)
Comment (retyui) [reply src/CameraRoll.ts]: no problem

**Review by Arjan-Zuidema — APPROVED** [review]

**Comment by retyui** [comment]
@Arjan-Zuidema any Ideas  how to fix android build ? regarding this [post](https://twitter.com/reactnative/status/1589296764678705155?s=20&t=RiLNXnCsVOG130PXjtN5Qw)

**Comment by Arjan-Zuidema** [comment]
Hmm, I thought you updated react-native to version 0.66.5? Looks like it got reverted somehow. Maybe you can try that 🙂

**Comment by retyui** [comment]
> Hmm, I thought you updated react-native to version 0.66.5? Looks like it got reverted somehow. Maybe you can try that slightly_smiling_face

yes, I tried to update and than I checked it didn't help, and I revered that change

**Comment by Arjan-Zuidema** [comment]
@retyui I got word there were some fixes for the react-native android artifacts on Maven Central. It should be fully resolved in a few hours. So I will see what I can do tomorrow

**Pull request merged**